src/main/java/com/redhat/sast/api/service/AbstractBatchService.java

```diff
@@ -0,0 +1,222 @@
+
+    protected final JobBatchRepository jobBatchRepository;
+    protected final JobService jobService;
+    protected final PlatformService platformService;
+    protected final PipelineParameterMapper parameterMapper;
+    protected final BatchProcessingService batchProcessingService;
+
+    protected AbstractBatchService(
+            JobBatchRepository jobBatchRepository,
+            JobService jobService,
+            PlatformService platformService,
+            PipelineParameterMapper parameterMapper,
+            BatchProcessingService batchProcessingService) {
+        this.jobBatchRepository = jobBatchRepository;
+        this.jobService = jobService;
+        this.platformService = platformService;
+        this.parameterMapper = parameterMapper;
+        this.batchProcessingService = batchProcessingService;
+    }
+
+    /**
+     * Retrieves a batch by its ID.
+     */
+    public JobBatchResponseDto getBatchById(@Nonnull Long batchId) {
+        JobBatch batch = jobBatchRepository.findById(batchId);
+        if (batch == null) {
+            throw new IllegalArgumentException("Batch not found with id: " + batchId);
+        }
+        return convertToResponseDto(batch);
+    }
+
+    /**
+     * Cancels all cancellable jobs in a batch.
+     */
+    @Transactional
+    public int cancelJobBatch(@Nonnull Long batchId) {
+        JobBatch batch = jobBatchRepository.findById(batchId);
+        if (batch == null) {
+            throw new IllegalArgumentException("Job batch not found with id: " + batchId);
+        }
+
+        List<Job> cancellableJobs = batch.getJobs().stream()
+                .filter(job -> canBeCancelled(job.getStatus()))
+                .toList();
+
+        if (cancellableJobs.isEmpty()) {
+            LOGGER.info("No cancellable jobs found in batch {}", batchId);
+            return 0;
+        }
+
+        LOGGER.info("Cancelling {} jobs in batch {}", cancellableJobs.size(), batchId);
+
+        int cancelledCount = 0;
+        for (Job job : cancellableJobs) {
+            try {
+                jobService.cancelJob(job.getId());
+                cancelledCount++;
+            } catch (Exception e) {
+                LOGGER.error("Failed to cancel job {} in batch {}", job.getId(), batchId, e);
+            }
+        }
+
+        batch.setStatus(BatchStatus.CANCELLED);
+        jobBatchRepository.persist(batch);
+
+        LOGGER.info("Cancelled {} out of {} jobs in batch {}", cancelledCount, cancellableJobs.size(), batchId);
+        return cancelledCount;
+    }
+
+    /**
+     * Determines if a job can be cancelled based on its status.
+     */
+    protected boolean canBeCancelled(JobStatus status) {
+        return status == JobStatus.RUNNING || status == JobStatus.SCHEDULED || status == JobStatus.PENDING;
+    }
+
+    /**
+     * Converts a JobBatch entity to a response DTO.
+     */
+    protected JobBatchResponseDto convertToResponseDto(@Nonnull JobBatch batch) {
```

Comment:
We don't need this method. We can utilize MapStruct to do this. We already have the dependency added into this project. Just create a mapper under mapper package and use it appropriately.